pool/pool_aio_test.go

```diff
@@ -108,14 +131,85 @@ func testEaclTable(containerID cid.ID) eacl.Table {
 	return table
 }
 
-func TestPoolInterfaceWithAIO(t *testing.T) {
+func TestPoolAio(t *testing.T) {
+	for _, version := range versions {
+		image := fmt.Sprintf("%s:%s", version.image, version.version)
+
+		t.Run(image, func(t *testing.T) {
+			t.Parallel()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			aioContainer := createDockerContainer(ctx, t, image)
+
+			// for instance: localhost:32781
+			nodeEndpoint, err := aioContainer.Endpoint(ctx, "")
+			require.NoError(t, err)
+
+			runTests(ctx, t, nodeEndpoint)
+
+			err = aioContainer.Terminate(ctx)
+			require.NoError(t, err)
+			cancel()
+			<-ctx.Done()
+		})
+	}
+}
+
+func runTests(_ context.Context, t *testing.T, nodeEndpoint string) {
+	nodeAddr := nodeAddress(nodeEndpoint)
+
+	t.Run("PoolInterfaceWithAIO", func(t *testing.T) {
+		testPoolInterfaceWithAIO(t, nodeAddr)
+	})
+
+	t.Run("PoolWaiterWithAIO", func(t *testing.T) {
+		testPoolWaiterWithAIO(t, nodeAddr)
+	})
+
+	t.Run("ClientWaiterWithAIO", func(t *testing.T) {
+		testClientWaiterWithAIO(t, nodeAddr)
+	})
+}
+
+func createDockerContainer(ctx context.Context, t *testing.T, image string) testcontainers.Container {
+	req := testcontainers.ContainerRequest{
+		Image:        image,
+		WaitingFor:   wait.NewLogStrategy("Serving neofs rest gw").WithStartupTimeout(45 * time.Second),
+		Name:         "sdk-poll-tests-" + strconv.FormatInt(time.Now().UnixNano(), 36),
```

Comment:
have you considered a date/time suffix as a better debug approach?

Reply:
It is just a container name, which you will never see, in general